BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c

```diff
@@ -380,6 +383,81 @@ UpdateMadt (
   return EFI_SUCCESS;
 }
 
+/**
+  This function updates FWST ACPI data structure based on ESRT table.
+
+  @param[in]  Current               Pointer to MADT table
+  @retval     EFI_SUCCESS           Patched the MADT
+              EFI_UNSUPPORTED       Sysinfo table not found.
+              EFI_OUT_OF_RESOURCES  If less MADT entries
+**/
+EFI_STATUS
+UpdateFwst (
+  IN UINT8   *Current
+  )
+{
+  UINT32                          RsvdBase;
+  UINT64                          OemId;
+  EFI_STATUS                      Status;
+  LOADER_GLOBAL_DATA              *LdrGlobal;
+  BOOT_LOADER_VERSION             *Version;
+  FIRMWARE_UPDATE_STATUS          *FwUpdateStatus;
+  EFI_FWST_ACPI_DESCRIPTION_TABLE *FwstAcpiTablePtr;
+
+  LdrGlobal = (LOADER_GLOBAL_DATA *)GetLoaderGlobalDataPointer();
+
+  //
+  // We do not need to populate ACPI table during firmware update
+  //
+  if (LdrGlobal->BootMode == BOOT_ON_FLASH_UPDATE) {
```

Comment:
Prefer using GetBootMode() instead of using LdrGlobal->BootMode.

Reply:
Done